models/classes/Lists/Business/Service/GetClassMetadataValuesService.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\Lists\Business\Service;
+
+use core_kernel_classes_Class;
+use core_kernel_classes_Property;
+use core_kernel_classes_Resource;
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\model\Lists\Business\Domain\Metadata;
+use oat\tao\model\Lists\Business\Domain\MetadataCollection;
+use oat\tao\model\Lists\Business\Domain\ValueCollectionSearchRequest;
+use oat\tao\model\Lists\Business\Input\ValueCollectionSearchInput;
+use tao_helpers_form_elements_Checkbox as CheckBox;
+use tao_helpers_form_elements_Combobox as ComboBox;
+use tao_helpers_form_elements_Htmlarea as HtmlArea;
+use tao_helpers_form_elements_Radiobox as RadioBox;
+use tao_helpers_form_elements_Textarea as TextArea;
+use tao_helpers_form_elements_Textbox as TextBox;
+use tao_helpers_form_elements_xhtml_Searchtextbox as SearchTextBox;
+
+class GetClassMetadataValuesService extends ConfigurableService
+{
+    use OntologyAwareTrait;
+
+    public const TEXT_WIDGETS = [
+        TextBox::WIDGET_ID,
+        TextArea::WIDGET_ID,
+        HtmlArea::WIDGET_ID,
+    ];
+    public const LIST_WIDGETS = [
+        RadioBox::WIDGET_ID,
+        CheckBox::WIDGET_ID,
+        ComboBox::WIDGET_ID,
+        SearchTextBox::WIDGET_ID,
+    ];
+    public const DATA_TYPE_LIST = 'list';
+    public const DATA_TYPE_TEXT = 'text';
+
+    private const BASE_LIST_ITEMS_URI = '/tao/PropertyValues/get?propertyUri=%s';
+
+    public function getByClass(
```

Comment:
I think this is more clear what is going on:

```suggestion
    public function getRecursivelyByClass(
```